tests/v2/validation/fleet/fleet.go

```diff
@@ -0,0 +1,155 @@
+	appsv1 "k8s.io/api/apps/v1"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	DeploymentResourceSteveType = "apps.deployment"
+	FleetClusterResourceType    = "fleet.cattle.io.cluster"
+	fleetControllerName         = "cattle-fleet-system/fleet-controller"
+)
+
+func verifyGitRepo(t *testing.T, client *rancher.Client, gitRepoID, k8sClusterID, steveClusterID string) {
+	backoff := kwait.Backoff{
+		Duration: 1 * time.Second,
+		Factor:   1.1,
+		Jitter:   0.1,
+		Steps:    20,
+	}
+	err := kwait.ExponentialBackoff(backoff, func() (finished bool, err error) {
+		client, err = client.ReLogin()
+		if err != nil {
+			return false, err
+		}
+
+		// after checking clusterStatus, check gitRepoStatus. gitRepoStatus starts in a healthy state, so if errors come up during clusterBundle deployments, its status will update to a negative / error state
+		gitRepo, err := client.Steve.SteveType(extensionsfleet.FleetGitRepoResourceType).ByID(gitRepoID)
+		if err != nil {
+			return false, err
+		}
+
+		gitStatus := &v1alpha1.GitRepoStatus{}
+		err = steveV1.ConvertToK8sType(gitRepo.Status, gitStatus)
+		if err != nil {
+			return false, err
+		}
+
+		if gitStatus.Display.Error {
+			return true, errors.New(gitStatus.Display.Message)
+		}
+
+		if gitRepo.State.Error {
+			return true, errors.New(gitRepo.State.Message)
+		}
+
+		logrus.Info(gitStatus.Summary)
+
+		if gitStatus.Summary.NotReady > 0 || gitStatus.Summary.DesiredReady == 0 || gitStatus.ReadyClusters == 0 {
+			return false, nil
+		}
+
+		return true, nil
+	})
+	require.NoError(t, err)
+
+	// this hits a currently known issue because the fleetCluster only updates every 15 mins. So unless we want the test to have this part be 15+ minutes, it will fail here for now. Possibly something for the team to fix?
+	logrus.Info("waiting for bundles to deploy to ", steveClusterID)
+	err = kwait.ExponentialBackoff(backoff, func() (finished bool, err error) {
+		client, err = client.ReLogin()
+		if err != nil {
+			return false, err
+		}
+
+		cluster, err := client.Steve.SteveType(FleetClusterResourceType).ByID(steveClusterID)
+		if err != nil {
+			return false, err
+		}
+
+		status := &v1alpha1.ClusterStatus{}
+		err = steveV1.ConvertToK8sType(cluster.Status, status)
+		if err != nil {
+			return false, err
+		}
+
+		logrus.Info(status.Summary.NonReadyResources)
+		for _, nonReadyResource := range status.Summary.NonReadyResources {
+			logrus.Info(nonReadyResource.Message)
+			if strings.Contains(nonReadyResource.Message, "error") || strings.Contains(nonReadyResource.Message, "Unable to continue") {
+				return true, errors.New(nonReadyResource.Message)
+			}
+		}
+
+		// after checking clusterStatus, check gitRepoStatus. gitRepoStatus can start in a healthy state, so if errors come up during clusterBundle deployments, its status will update to a negative / error state that aren'
+		gitRepo, err := client.Steve.SteveType(extensionsfleet.FleetGitRepoResourceType).ByID(gitRepoID)
```

Comment:
I'm a bit confused. why do we want to verify the gitRepo status again as we have already verified its status in the previous lines 40 through 65?

Reply:
`gitRepoStatus can start in a healthy state, so if errors come up during clusterBundle deployments, its status will update to a negative / error state that aren't caught right now in the clusterBundle due to a timing issue in fleet:` https://github.com/rancher/fleet/issues/2699

basically if an error is hit, clusterBundle isn't updated in a timely manner. So it makes more sense to check both clusterBundle and gitRepo status' in order to check that the bundle is deploying correctly.